src/p4est_io.c

```diff
@@ -469,22 +469,22 @@ check_file_metadata (sc_MPI_Comm mpicomm, const char *filename,
   }
 
   /* check the format of the user string */
-  if (metadata[47] != '\n') {
+  if (metadata[79] != '\n') {
     if (rank == 0) {
       P4EST_LERROR (P4EST_STRING
                     "_io: Error reading. Wrong file header format.\n");
     }
     return P4EST_ERR_IO;
   }
   /* the content of the user string is not checked */
-  strncpy (user_string, &metadata[32], 15);
-  user_string[15] = '\0';
+  strncpy (user_string, &metadata[32], P4EST_NUM_USER_STRING_BYTES - 1);
+  user_string[P4EST_NUM_USER_STRING_BYTES - 1] = '\0';
 
   /* check number of global quadrants */
   /* there is no \n at the end of this line */
   metadata[P4EST_NUM_METADATA_BYTES] = '\0';
 
-  if (strlen (&metadata[48]) != 16) {
+  if (strlen (&metadata[80]) != 16) {
```

Comment:
The positions 32 and 80 can be changed to be the sum of `#define`s such that we may change the values.

Reply:
Yes, see also the related comment of the review above.